tycho-packaging-plugin/src/main/java/org/eclipse/tycho/buildversion/ValidateIdMojo.java

```diff
@@ -17,17 +17,23 @@
 import org.apache.maven.plugin.MojoFailureException;
 import org.apache.maven.plugins.annotations.LifecyclePhase;
 import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
 import org.eclipse.tycho.PackagingType;
 
 /**
  * Validates that project Maven and OSGi ids match.
  */
 @Mojo(name = "validate-id", defaultPhase = LifecyclePhase.VALIDATE, threadSafe = true)
 public class ValidateIdMojo extends AbstractVersionMojo {
+    /**
+     * Whether to skip the project's artifact ID validation against the OSGi ID.
+     */
+    @Parameter
+    private boolean skip;
```

Comment:
This is not really required, but a way to skip the validation to apply some "hacks" might be useful.